tests/console/year_2038_detection.pm

```diff
@@ -34,43 +37,42 @@ sub install_pkg {
 }
 
 sub run {
+    my $self = shift;
     select_serial_terminal;
 
     install_pkg if (script_run('rpm -qi chrony') != 0);
     systemctl("start chronyd");    # Ensure chrony is started
-    assert_script_run 'chronyc makestep';
+    assert_script_run('chronyc makestep');
     record_info('Show current date and time', script_output('date +"%Y-%m-%d"'));
     assert_script_run('utmpdump /var/run/utmp');
     assert_script_run('utmpdump /var/log/wtmp');
 
     # Stop the chrony service so that we can change date and time
-    systemctl('stop chronyd.service');
+    disable_and_stop_service('chronyd.service');
+
     # Set the time and date beyond a Y2038
+    record_info('Set new date');
     assert_script_run('timedatectl set-time "2038-01-20 03:14:07"');
 
     # We may need to logout and login again to make the date/time change
-    # take effect. in a simple way, we can switch to another user to
-    # achieve this.
-    #
-    # Create user account, if image doesn't already contain user
-    # (which is the case for SLE images that were already prepared by openQA)
-    if (script_run("getent passwd $username") != 0) {
-        assert_script_run "useradd -m $testapi::username";
-        assert_script_run "echo '$testapi::username:$testapi::password' | chpasswd";
+    # However, we need to handle many platforms and different products.
+    # Reboot the system to acieve this is a simple way for the time being.
+    if (is_transactional) {
+        process_reboot(trigger => 1);
     }
-    ensure_serialdev_permissions;
-
-    # Switch user to check if the issue can be captured
-    select_console 'user-console';
+    else {
+        power_action('reboot', textmode => 1);
+        reconnect_mgmt_console if is_pvm;
+        $self->wait_boot(textmode => 1, ready_time => 600, bootloader_time => 300);
+    }
+    select_serial_terminal;
```

Comment:
I would add an additional `record_info` here, to help us debug time issues after the reboot, e.g.

```
record_info("time after reboot", script_output("timedatectl status"));
```